### PR TITLE
Updated for NATS 2.0, added README for benchmarks

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: examples/stan-bench/main.go [hunk 1]
```diff
@@ -31,17 +30,17 @@ import (
 // Some sane defaults
 const (
 	DefaultNumMsgs            = 100000
-	DefaultNumPubs            = 1
+	DefaultNumPubs            = 10
 	DefaultNumSubs            = 0
-	DefaultAsync              = false
+	DefaultAsync              = true
```
Comment (kozlovic) [inline examples/stan-bench/main.go]: So it means that user would have to pass `-a=false` now to publish sync. Should we add `-sync` to make it easier?

File: examples/stan-sub/main.go [hunk 2]
```diff
@@ -29,19 +30,19 @@ var usageStr = `
...
-	         (for more information: https://golang.org/pkg/time/#ParseDuration)
-	--durable <name>                Durable subscriber name
-	--unsubscribe                   Unsubscribe the durable on exit
+	--qgroup <name>                  Queue group
+	--all                            Deliver all available messages
+	--last                           Deliver starting with last published message
+	--since  <time_ago>              Deliver messages in last interval (e.g. 1s, 1hr)
+	--seq    <seqno>                 Start at seqno
+	--durable <name>                 Durable subscriber name
+	--unsub                          Unsubscribe the durable on exit
```
Comment (kozlovic) [inline examples/stan-sub/main.go]: You replaced `--unsubscribe` with `-unsub` here, but not in the parser. Fine to have `--unsub` here, but then add (not replace) it in the parser.

File: examples/stan-sub/main.go [hunk 3]
```diff
@@ -50,38 +51,38 @@ func usage() {
...
 	flag.StringVar(&clusterID, "cluster", "test-cluster", "The NATS Streaming cluster ID")
-	flag.StringVar(&clientID, "id", "", "The NATS Streaming client ID to connect with")
-	flag.StringVar(&clientID, "clientid", "", "The NATS Streaming client ID to connect with")
+	flag.StringVar(&clientID, "id", "stan-sub", "The NATS Streaming client ID to connect with")
+	flag.StringVar(&clientID, "clientid", "stan-sub", "The NATS Streaming client ID to connect with")
 	flag.BoolVar(&showTime, "t", false, "Display timestamps")
 	// Subscription options
 	flag.Uint64Var(&startSeq, "seq", 0, "Start at sequence no.")
-	flag.BoolVar(&deliverAll, "all", false, "Deliver all")
+	flag.BoolVar(&deliverAll, "all", true, "Deliver all")
```
Comment (kozlovic) [inline examples/stan-sub/main.go]: By default you would get new-only, but now, without any arg, you would get them all. To have new-only you would then need `-all=false`. Should we add `-new_only` or something?

File: examples/stan-bench/main.go (hunk 1 above)
Comment (derekcollison) [reply examples/stan-bench/main.go]: Yes I agree. Will fix.

File: examples/stan-sub/main.go (hunk 2 above)
Comment (derekcollison) [reply examples/stan-sub/main.go]: Will fix.

File: examples/stan-sub/main.go (hunk 3 above)
Comment (derekcollison) [reply examples/stan-sub/main.go]: Let me look.

Comment (derekcollison) [reply examples/stan-sub/main.go]: Will fix.

File: examples/stan-sub/main.go [hunk 4]
```diff
@@ -50,56 +52,68 @@ func usage() {
...
+	flag.BoolVar(&newOnly, "new_only", false, "Only new messages")
 	flag.BoolVar(&deliverLast, "last", false, "Start with last value")
 	flag.StringVar(&startDelta, "since", "", "Deliver messages since specified time offset")
 	flag.StringVar(&durable, "durable", "", "Durable subscriber name")
-	flag.StringVar(&qgroup, "qgroup", "", "Queue group name")
-	flag.BoolVar(&unsubscribe, "unsubscribe", false, "Unsubscribe the durable on exit")
+	flag.StringVar(&qgroup, "qgroup", "stan", "Queue group name")
+	flag.BoolVar(&unsubscribe, "unsub", false, "Unsubscribe the durable on exit")
```
Comment (kozlovic) [inline examples/stan-sub/main.go]: I would have left the old `unsubscribe`, basically having both support for -unsub and -unsubscribe map to the unsubscribe boolean.

File: examples/stan-bench/main.go [hunk 5]
```diff
@@ -31,17 +30,17 @@ import (
...
 	DefaultMessageSize        = 128
 	DefaultIgnoreOld          = false
-	DefaultMaxPubAcksInflight = 1000
+	DefaultMaxPubAcksInflight = 512
 	DefaultClientID           = "benchmark"
 )
 
 func usage() {
-	log.Fatalf("Usage: stan-bench [-s server (%s)] [-tls] [-c CLUSTER_ID] [-id CLIENT_ID] [-qgroup QUEUE_GROUP_NAME] [-np NUM_PUBLISHERS] [-ns NUM_SUBSCRIBERS] [-n NUM_MSGS] [-ms MESSAGE_SIZE] [-csv csvfile] [-mpa MAX_NUMBER_OF_PUBLISHED_ACKS_INFLIGHT] [-io] [-a] <subject>\n", nats.DefaultURL)
+	log.Fatalf("Usage: stan-bench [-s server (%s)] [-c CLUSTER_ID] [-id CLIENT_ID] [-qgroup QUEUE_GROUP_NAME] [-np NUM_PUBLISHERS] [-ns NUM_SUBSCRIBERS] [-n NUM_MSGS] [-ms MESSAGE_SIZE] [-csv csvfile] [-mpa MAX_NUMBER_OF_PUBLISHED_ACKS_INFLIGHT] [-io] [-sync] <subject>\n", nats.DefaultURL)
```
Comment (kozlovic) [inline examples/stan-bench/main.go]: I don't see the `-creds` here.

File: examples/stan-sub/main.go [hunk 6]
```diff
@@ -50,56 +52,68 @@ func usage() {
...
-	var durable string
-	var qgroup string
-	var unsubscribe bool
-	var URL string
-
-	//	defaultID := fmt.Sprintf("client.%s", nuid.Next())
+	var (
+		clusterID, clientID string
+		URL                 string
+		userCreds           string
```
Comment (kozlovic) [inline examples/stan-sub/main.go]: I don't see the creds added in the flag parsing?

File: examples/stan-sub/main.go (hunk 4 above)
Comment (derekcollison) [reply examples/stan-sub/main.go]: Will fix.

File: examples/stan-bench/main.go (hunk 5 above)
Comment (derekcollison) [reply examples/stan-bench/main.go]: Will fix.

File: examples/stan-sub/main.go (hunk 6 above)
Comment (derekcollison) [reply examples/stan-sub/main.go]: Added.

**Pull request merged**